src/platform/tls_openssl.c

```diff
@@ -89,6 +89,11 @@ typedef struct QUIC_TLS {
 
 } QUIC_TLS;
 
+typedef struct QUIC_HP_KEY {
+    EVP_CIPHER_CTX* CipherCtx;
```

Comment:
Is there, by chance, a way to trivially query the AEAD type from this pointer? That way we wouldn't need the extra object/indirection.

Reply:
It's unlikely; the struct is opaque by design.  There might be an API to query the object type, but that would probably be slower than a double indirection, wouldn't it?